src/Composer/Util/RemoteFilesystem.php

```diff
@@ -436,6 +449,9 @@ protected function getOptionsForUrl($originUrl, $additionalOptions)
             $options['http']['header'][] = $header;
         }
 
+        if($this->config && $this->config->get('gitlab-domains') && $originUrl == $this->config->get('gitlab-domains')[0]) {
+            $options['retry-auth-failure'] = false;
```

Comment:
By removing a condition in `GitLab` class, this line is no longer necessary.
https://github.com/GromNaN/composer/commit/db056b838341a6dc6822fced57c2320e4082460f#diff-05f3e1829e608fea85d7b258821d100cL125